ext/proj4_c_impl/main.c

```diff
@@ -53,24 +59,51 @@ static void rgeo_proj4_free(void *ptr)
 
 static size_t rgeo_proj4_memsize(const void *ptr)
 {
+  size_t size = 0;
   const RGeo_Proj4Data *data = (const RGeo_Proj4Data *)ptr;
-  return sizeof(*data);
+
+  size += sizeof(*data);
+  if(data->pj){
+    size += sizeof(data->pj);
+  }
+  return size;
 }
 
 static void rgeo_proj4_mark(void *ptr)
 {
   RGeo_Proj4Data *data = (RGeo_Proj4Data *)ptr;
   if(!NIL_P(data->original_str)){
-    rb_gc_mark(data->original_str);
+    mark(data->original_str);
   }
 }
 
+#ifdef HAVE_RB_GC_MARK_MOVABLE
+static void rgeo_proj4_compact(void *ptr)
+{
+  RGeo_Proj4Data *data = (RGeo_Proj4Data *)ptr;
+  if(data && !NIL_P(data->original_str)){
```

Comment:
I'm curious if this `NIL_P` conditional is necessary or could even be dangerous. If the string were moved, would calling this function with the old pointer even work?

Reply:
After looking into this a little more, I think this is fine. If `original_str` were moved, then instead of `T_STRING` it would point to a `T_MOVED` value, which should respond properly to `NIL_P`.